src/extra/others/fragment/lv_fragment.h

```diff
@@ -0,0 +1,232 @@
+/**
+ * @file lv_fragment.h
+ *
+ */
+
+#ifndef lv_fragment_H
+#define lv_fragment_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "../../../lv_conf_internal.h"
+
+#if LV_USE_FRAGMENT
+
+#include "../../../core/lv_obj.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+typedef struct _lv_fragment_manager_t lv_fragment_manager_t;
+
+typedef struct lv_fragment_t lv_fragment_t;
+typedef struct lv_fragment_class_t lv_fragment_class_t;
+
+/**
+ * Opaque pointer for internal state management
+ */
+typedef struct manager_stack_t manager_stack_t;
+
+struct lv_fragment_t {
+    /**
+     * Class of this fragment
+     */
+    const lv_fragment_class_t * cls;
+    /**
+     * Manager this fragment instance belongs to
+     */
+    lv_fragment_manager_t * manager;
+    /**
+     * lv_obj returned by create_obj_cb
+     */
+    lv_obj_t * obj;
```

Comment:
Is it for the user in case he wants to store an obj?

Reply:
There are 2 cases.
1: User returns obj, and it will be managed along with the fragment
2: User doesn't return obj, means they will have to manage the views they added manually

For case 2 this is null.